contracts/Marketplace.sol

```diff
@@ -298,6 +317,9 @@ contract Marketplace is SlotReservations, Proofs, StateRetrieval, Endian {
     _removeFromMySlots(slot.host, slotId);
     uint256 slotIndex = slot.slotIndex;
     delete _slots[slotId];
+    delete _reservations[slotId]; // We purge all the reservations for the slot
+    slot.state = SlotState.Repair;
+    slot.requestId = requestId;
```

Comment:
Why is this needed? Shouldn't it have been already set in `fillSlot`?

Reply:
`delete _slots[slotId]` will delete all the info ;-) so I am just repopulating required data here.

We need the `requestId` otherwise the `slotState()` will report `Free` even though `slot.state` is set.